Source/SwiftLintFramework/Rules/Lint/RequiredXCTestTearDownRule.swift

```diff
@@ -0,0 +1,122 @@
+import SourceKittenFramework
+
+public struct RequiredXCTestTearDownRule: Rule, OptInRule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public static let description = RuleDescription(
+        identifier: "required_xctest_teardown",
+        name: "Required XCTest Tear Down",
+        description: "Test classes must implement tearDown when setUp is provided.",
+        kind: .lint,
+        nonTriggeringExamples: [
```

Comment:
can you add some examples using the [`class func setUp`](https://developer.apple.com/documentation/xctest/xctestcase/1496262-setup) and [`class func tearDown`](https://developer.apple.com/documentation/xctest/xctestcase/1496280-teardown)?

I wonder if it makes sense to check that they're balanced (e.g. a `class setUp` requires a `class tearDown`)

Reply:
Added :-) It does make sense to check if they're balanced. Since they are static methods they have potential to leak to other tests if misused.